### PR TITLE
#1512 Add QR scanner on withdraw (beta)

### PR DESCRIPTION
**Pull request by gibbsfromncis**
**Tested on MacOS:**

- [x] Firefox
- [x] Chrome
- [x] Opera

**Issue:** #1512

QR scanner tries to parse QR code as string "bitcoint:(address)?amount=(quantity)". If it doesn't match regexp  - it uses a whole string as bitcoin address. If string matched scanner parses address & amount.

_I had no ability to test it on my phone because when I try to open wallet on my iPhone I always redirected to init-error page._ 

There are some screens how it looks like:

screenshot 2018-05-21 11 53 02
screenshot 2018-05-21 11 47 15

### CODE REVIEW

File: app/components/QRAddressScanner.jsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                                width: "calc(100% - 48px)",
+                                margin: "0 24px"
+                            }}
+                        />
+
+                        {this.state.address && (
+                            <div>
+                                <div className="qr-address-scanner-status">
+                                    <div className="qr-address-scanner-status-title">
+                                        Address found:
```
Comment (svk31) [inline app/components/QRAddressScanner.jsx]: You should add translation keys for all text entries. <Translate content="new_key" />

File: app/components/Modal/WithdrawModalNew.jsx [hunk 2]
```diff
@@ -838,6 +840,19 @@ class WithdrawModalNew extends React.Component {
         }
     }
 
+    handleQrScanSuccess(data) {
+        if (!this.state.quantity) {
```
Comment (svk31) [inline app/components/Modal/WithdrawModalNew.jsx]: Shouldn't this be checking if `data` contains an `amount` field?

Comment (gibbsfromncis) [reply app/components/Modal/WithdrawModalNew.jsx]: @svk31 This if checks if user put something on "quantity" field on modal by himself. If yes I do not put amount detected on QR code. If user don't put "quantity" by himself I put amount to quantity. I put a comment for this if.

**Comment by gibbsfromncis** [comment]
@svk31 done. I'm sorry for my inattention this time.

**Comment by svk31** [comment]
Tested on Galaxy S9+ and it works. It did output a massive Chinese text at one point when laid flat on my desk, think it was an error message? Because of the "else" clause in `onScanSuccess` that text got set as a possible address. The user still has to accept that though so it shouldn't be a big problem. As we start seeing real use cases we might want to tweak the data handling in onScanSuccess however.

**Pull request merged**